docuware-connector/src/com/axonivy/connector/docuware/connector/DocuWareService.java

```diff
@@ -55,4 +95,139 @@ public Date stringToDate(String dateString) {
 		}
 		return date;
 	}
+	
+	public static Document uploadFile(WebTarget target, java.io.File file, DocuWareEndpointConfiguration configuration, List<DocuWareProperty> properties) throws IOException, DocuWareException {
+		byte[] bytes = Files.readAllBytes(file.toPath());
+		return uploadStream(target,bytes, file.getName(), configuration, properties);
+	}
+	
+	public static Document uploadStream(WebTarget target, byte[] file, String fileName, DocuWareEndpointConfiguration configuration, List<DocuWareProperty> properties) throws IOException, DocuWareException {
+		FormDataMultiPart multipart;
+		File propertiesFile = createPropertiesFile(properties);
+		
+		try (FormDataMultiPart formDataMultiPart = new FormDataMultiPart()) {		
+			InputStream streamProperties = new FileInputStream(propertiesFile.getJavaFile());
+			StreamDataBodyPart streamPropertiesPart = new StreamDataBodyPart(PROPERTIES_FILE_NAME, streamProperties );
+			streamPropertiesPart.setMediaType(MediaType.APPLICATION_JSON_TYPE);
```

Comment:
this is ok for now: but let's clarify why we need a special uploader today:
- [ ] the property-order of the form is important ... before 10.0.2 we could not guarantee sending order.
- [ ] the byte[] natured payload is easier to handle than files. 
is that correct or anything else involved?

Reply:
besides the issue with the property order and the bye[] payload one another reason was that if we want to use a json file for the document-index update we can not define the content-type (application/json) for this file (only for this index file) in the rest client call activity. Without the content-type information the upload did not work properly.